plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedOnceRetentionStrategy.java

```diff
@@ -24,6 +24,8 @@
 
     private boolean terminating;
 
+    private boolean tempOffline;
```

Comment:
Better make it volatile or use synchronization.

Reply:
I though sync isn't needed in this case, so I'll make it volatile.